// @clxx/layout/build/Col.js
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import { jsx } from "@emotion/core";
import { getFlexAxisStyles } from "./getFlexAxisStyles";
export function Col(props) {
    const { wrap, justify, align } = props, attributes = __rest(props, ["wrap", "justify", "align"]);
    const styles = {
        display: "flex",
        flexDirection: "column",
        flexWrap: wrap,
        justifyContent: justify,
        alignItems: align
    };
    return (jsx("div", Object.assign({ css: styles }, attributes), props.children));
}
export function ColAround(props) {
    const { styles, attributes } = getFlexAxisStyles(props, "column", "space-around");
    return (jsx("div", Object.assign({ css: styles }, attributes), props.children));
}
export function ColCenter(props) {
    const { styles, attributes } = getFlexAxisStyles(props, "column", "center");
    return (jsx("div", Object.assign({ css: styles }, attributes), props.children));
}
export function ColBetween(props) {
    const { styles, attributes } = getFlexAxisStyles(props, "column", "space-between");
    return (jsx("div", Object.assign({ css: styles }, attributes), props.children));
}
export function ColStart(props) {
    const { styles, attributes } = getFlexAxisStyles(props, "column", "flex-start");
    return (jsx("div", Object.assign({ css: styles }, attributes), props.children));
}
export function ColEnd(props) {
    const { styles, attributes } = getFlexAxisStyles(props, "column", "flex-end");
    return (jsx("div", Object.assign({ css: styles }, attributes), props.children));
}
