// @clxx/layout/build/FixContainer.js
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import { jsx } from "@emotion/core";
export function FixContainer(props) {
    const { showMask = true, centerChild = true, children } = props, attributes = __rest(props, ["showMask", "centerChild", "children"]);
    const styles = {
        position: "fixed",
        left: "50%",
        zIndex: 9,
        width: "100%",
        maxWidth: "576px",
        transform: `translateX(-50%)`,
        height: "100%",
        top: 0
    };
    if (showMask) {
        styles.backgroundColor = `rgba(0,0,0,.6)`;
    }
    if (centerChild) {
        styles.display = "flex";
        styles.justifyContent = "center";
        styles.alignItems = "center";
    }
    return (jsx("div", Object.assign({ css: styles }, attributes), children));
}
